packages/core/src/config/parse.ts

```diff
@@ -676,21 +832,119 @@ const parseContractVariables = (
+      message += `The following variables were defined in the contract ${contractConfig.contract} (or one of its parent contracts) but were not defined in the ChugSplash config file:\n`
+      for (const variable of missingVariables) {
+        message += `${variable} \n`
+      }
+      message += `- Every variable defined in your contracts must be assigned a value in your ChugSplash config file.\n`
+      message += `- Please define the variable in your ChugSplash config file then run this command again.\n`
+      message += `- If this problem persists, delete your cache folder then try again.\n\n`
+    }
+
     throw new Error(message)
   }
 
   return parsedConfigVariables
 }
 
+/**
+ * Parses and validates constructor args in a config file.
+ *
+ * TODO - Improve this function so that the parsing and validation is on par with that of variables.
+ *
+ * @param contractConfig Unparsed User-defined contract definition in a ChugSplash config.
+ * @param storageLayout Storage layout returned by the solidity compiler for the relevant contract.
+ * @param compilerOutput Complete compiler output.
+ * @returns complete set of variables parsed into the format expected by the parsed chugsplash config.
+ */
+export const parseAndValidateConstructorArgs = (
+  userConstructorArgs: UserConfigVariables,
+  referenceName: string,
+  abi: Array<Fragment>
+): ParsedConfigVariables => {
+  const parsedConstructorArgs: ParsedConfigVariables = {}
+
+  const constructorFragment = abi.find(
+    (fragment) => fragment.type === 'constructor'
+  )
+
+  if (constructorFragment === undefined) {
+    if (Object.keys(userConstructorArgs).length > 0) {
+      throw new InputError(
+        `User entered constructor arguments in the ChugSplash config file for ${referenceName}, but\n` +
+          `no constructor exists in the contract.`
+      )
+    } else {
+      return parsedConstructorArgs
+    }
+  }
+
+  if (
+    Object.keys(userConstructorArgs).length > constructorFragment.inputs.length
+  ) {
+    const constructorArgNames = constructorFragment.inputs.map(
+      (input) => input.name
+    )
+    const incorrectConstructorArgNames = Object.keys(
+      userConstructorArgs
+    ).filter((argName) => !constructorArgNames.includes(argName))
+    throw new Error(
+      `User entered an incorrect number of constructor arguments in the ChugSplash config file for ${referenceName}.\n` +
+        `Please remove the following variables from the 'constructorArgs' field:` +
+        `${incorrectConstructorArgNames.map((argName) => `\n${argName}`)}`
+    )
+  }
+
+  constructorFragment.inputs.forEach((input) => {
+    const constructorArgValue = userConstructorArgs[input.name]
+    if (constructorArgValue === undefined) {
+      throw new Error(
+        `User did not define the constructor argument '${input.name}' in the ChugSplash config file\n` +
+          `for ${referenceName}. Please include it in the 'constructorArgs' field in your ChugSplash config file.`
```

Comment:
Would be great to have the list of missing constructor args instead of throwing an error for the first missing one